gemrb/GUIScripts/PortraitWindow.py

```diff
@@ -0,0 +1,362 @@
+		button.SetAction (GUICommonWindows.PortraitButtonOnShiftPress, IE_ACT_MOUSE_PRESS, GEM_MB_ACTION, GEM_MOD_CTRL, 1)
+		button.SetAction (GUICommonWindows.ButtonDragSourceHandler, IE_ACT_DRAG_DROP_SRC)
+		button.SetAction (GUICommonWindows.ButtonDragDestHandler, IE_ACT_DRAG_DROP_DST)
+
+		pairs[i] = button
+		limit -= limitStep
+
+	# move the buttons back up, to combine the freed space
+	if scale:
+		for i in range(PartySize - 1):
+			button = pairs[i]
+			button.SetSize (buttonWidth, buttonHeight)
+			if i == 0:
+				continue # don't move the first portrait
+			rect = button.GetFrame ()
+			x = rect["x"]
+			y = rect["y"]
+			button.SetPos (x, y-portraitGap*i)
+
+	return pairs
+
+def AddStatusFlagLabel (Window, Button, i):
+	label = Window.GetControl (200 + i)
+	if label:
+		return label
+
+	# label for status flags (dialog, store, level up)
+	align = IE_FONT_ALIGN_TOP | IE_FONT_ALIGN_RIGHT | IE_FONT_SINGLE_LINE
+	if GameCheck.IsBG2 ():
+		align = IE_FONT_ALIGN_TOP | IE_FONT_ALIGN_CENTER | IE_FONT_SINGLE_LINE
+	label = Button.CreateLabel (200 + i, StatesFont, "", align) #level up icon is on the right
+	label.SetFrame (Button.GetInsetFrame (4))
+	return label
+
+# overlay a label, so we can display the hp with the correct font. Regular button label
+#   is used by effect icons
+def AddHPLabel (Window, Button, i):
+	label = Window.GetControl (100 + i)
+	if label:
+		return label
+
+	label = Button.CreateLabel (100 + i, "NUMFONT", "", IE_FONT_ALIGN_TOP | IE_FONT_ALIGN_LEFT | IE_FONT_SINGLE_LINE)
+	return label
+
+def SetupButtonBorders (Window, Button, i):
+	# unlike other buttons, this one lacks extra frames for a selection effect
+	# so we create it and shift it to cover the grooves of the image
+	# except iwd2's second frame already has it incorporated (but we miscolor it)
+	yellow = {'r' : 255, 'g' : 255, 'b' : 0, 'a' : 255}
+	green = {'r' : 0, 'g' : 255, 'b' : 0, 'a' : 255}
+	if GameCheck.IsIWD2 ():
+		Button.SetBorder (FRAME_PC_SELECTED, green)
+		Button.SetBorder (FRAME_PC_TARGET, yellow, 0, 0, Button.GetInsetFrame (2, 2, 3, 3))
+	else:
+		Button.SetBorder (FRAME_PC_SELECTED, green, 0, 0, Button.GetInsetFrame (4, 3, 4, 3))
+		Button.SetBorder (FRAME_PC_TARGET, yellow, 0, 0, Button.GetInsetFrame (2, 2, 3, 3))
+
+def MinimizePortraits(): #bg2
+	GemRB.GameSetScreenFlags(GS_PORTRAITPANE, OP_OR)
+
+def OpenPortraitWindow (needcontrols=0, pos=WINDOW_RIGHT|WINDOW_VCENTER):
+	#take care, this window is different in how/iwd
+	if GameCheck.HasHOW() and needcontrols:
+		Window = GemRB.LoadWindow (26, GUICommon.GetWindowPack(), pos)
+	else:
+		Window = GemRB.LoadWindow (1, GUICommon.GetWindowPack(), pos)
+
+	Window.AddAlias("PORTWIN")
+	Window.SetFlags(WF_BORDERLESS|IE_GUI_VIEW_IGNORE_EVENTS, OP_OR)
+
+	if needcontrols:
+		# 1280 and higher don't have this control
+		Button = Window.GetControl (8)
+		if Button:
+			if GameCheck.IsIWD():
+				# Rest (iwd)
+				Button.SetTooltip (11942)
+				Button.OnPress (RestPress)
```

Comment:
Ok, I left my comment at the wrong `PortraitWindow.py` during investigation.

_This_ is one of the three trouble-causing `RestPress`.